sample/a-buffer/main.ts

```diff
@@ -13,8 +13,12 @@ function roundUp(n: number, k: number): number {
 }
 
 const canvas = document.querySelector('canvas') as HTMLCanvasElement;
-const adapter = await navigator.gpu?.requestAdapter();
-const device = await adapter?.requestDevice();
+const adapter = await navigator.gpu?.requestAdapter({
+  featureLevel: 'compatibility',
+});
+const device = await adapter?.requestDevice({
+  requiredLimits: { maxStorageBuffersInFragmentStage: 2 },
+});
```

Comment:
@kainino0x pointed out this will fail on current chrome/firefox/safari

To make it work requires only adding it IF `maxStorageBuffersInFragmentStage` exists on `adapter.limits` because otherwise it's an unknown limit and will fail device creation.

There's a bunch of solutions - they all have trade offs.

Here's one

```suggestion
const adapter = await navigator.gpu?.requestAdapter({
  featureLevel: 'compatibility',
});
const { maxStorageBuffersInFragmentStage } = adapter.limits;

// Note: maxStorageBuffersInFragmentStage may be undefined in old browsers
// but that's ok as it will fail this check and old browser only support core
// which always has enough storage buffers in fragment shaders.
if (maxStorageBuffersInFragmentStage < 2) {
  fail('maxStorageBuffersInFragmentStage < 2'); // export from util.ts
}
const device = await adapter?.requestDevice({
  requiredLimits: { maxStorageBuffersInFragmentStage },
);
```

Reply:
Note: I'm not sure that example is acceptable nor not since it's not requesting 2, it's requesting max. The more I have to deal with features and limits the more think you should just always get max limits. It's much easier to deal with.

```js
const adapter = await navigator.gpu?.requestAdapter({
  featureLevel: 'compatibility',
});
const device = await adapter.requestDevice({
  requiredFeatures: adapter.features,
  requiredLimits: objLikeToObj(adapter.limits), // need a helper ATM
});
if (device.limits.maxStorageBuffersInFragmentStage < 2) {
  fail('I can has no running');
}
```

It's way simpler than the summersault of checking for in adapter and applying in device but it's not the intent of the spec.

Reply:
Thanks.

> const { maxStorageBuffersInFragmentStage } = adapter.limits;

gave me:

```
(!) [plugin typescript] sample/a-buffer/main.ts (20:9): @rollup/plugin-typescript TS2339: Property 'maxStorageBuffersInFragmentStage' does not exist on type 'GPUSupportedLimits'.
/home/senorblanco/src/webgpu-samples/sample/a-buffer/main.ts:20:9
```

I used 

```
if ('maxStorageBuffersInFragmentStage' in adapter.limits) {
```

instead. LMK if that's kosher.

Reply:
> Note: I'm not sure that example is acceptable nor not since it's not requesting 2, it's requesting max. The more I have to deal with features and limits the more think you should just always get max limits. It's much easier to deal with.

It is easier, but I think it's dangerous, in that it turns WebGPU into OpenGL: you could write some code that runs fine on your machine (with above-average limits), but won't run elsewhere (with default limits). I'd prefer to request the minimum that the sample actually needs.

Reply:
> const { maxStorageBuffersInFragmentStage } = adapter.limits;

It gets an error because the types need to be updated

```
npm install --save-dev @webgpu/types@latest
```